File: 1/1.go
func twoSum(nums []int, target int) []int {
	cache := make(map[int]int)

	for i, val := range nums {
		// Look for corresponding number
		if j, ok := cache[target - val]; ok {
			return []int {j, i}
		}

		// Not found, push this number into the map
		cache[val] = i
	}

	return []int {-1, -1}
}

File: 1-99/1/1.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
	vector<int> twoSum(vector<int>& nums, int target) {
		unordered_map<int, int> cache;
		
		for (int i = 0; i < nums.size(); i++) {
			// Found in cache, return the indice
			if (cache.find(target - nums[i]) != cache.end())
				return vector<int> {cache[target - nums[i]], i};
			
			// Not found in cache, add index to cache
			cache[nums[i]] = i;
		}
		
		return vector<int> {-1, -1};
	}
};